fail2ban/server/filter.py

```diff
@@ -844,35 +943,50 @@ def getJournalMatch(self, match): # pragma: no cover - Base class, not used
 
 import socket
 import struct
+from .utils import Utils
 
 
 class DNSUtils:
 
 	IP_CRE = re.compile("^(?:\d{1,3}\.){3}\d{1,3}$")
 
+	# todo: make configurable the expired time and max count of cache entries:
+	CACHE_dnsToIp = Utils.Cache(maxCount=1000, maxTime=60*60)
```

Comment:
> what about at most 5*60

Agree, but I will make it configurable (fail2ban.conf), so we can set default to `1000` (size) and `5m` (time)

About the test cases: I will extend the test cases for all that also - that was a first shot, directly after chery pick and rebase.